### PR TITLE
Activation 2.1: Split the Eclipse Angus Reference Implementation into separate bundle

### PR DESCRIPTION
**Pull request by neuwerk**
This pull requests splits the Activation 2.1 RI into a separate bundle, now that the specification APIs are no longer dependent on the RI.

### CODE REVIEW

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
@jhanders34 if you wouldn't mind reviewing this for me, one question I had is if its alright to leave the RI bundle in the what could become just the spec private feature.

File: dev/io.openliberty.org.eclipse.angus.activation/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Copyright (c) 2023 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-2.0/
+# 
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.org.eclipse.angus.activation/bnd.bnd]:
```suggestion
```

We are not including this in new files.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

**Comment by neuwerk** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.restfulWS-3.1.feature [hunk 2]
```diff
@@ -8,7 +8,6 @@ IBM-Process-Types: client, \
  server
 Subsystem-Name: Jakarta RESTful Web Services 3.1
 -features=com.ibm.websphere.appserver.eeCompatible-10.0, \
-  io.openliberty.jakarta.activation-2.1, \
   io.openliberty.jakarta.xmlBinding-4.0; apiJar=false
 -bundles=\
   io.openliberty.jaxrs30; location:="dev/api/ibm/,lib/", \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.restfulWS-3.1.feature]:
```suggestion
```
Also `-files` .  Both should be moved to restfulWSClient-3.1 public feature.

File: dev/com.ibm.websphere.appserver.features/visibility/public/mail-2.1/io.openliberty.mail-2.1.feature [hunk 3]
```diff
@@ -8,6 +8,7 @@ Subsystem-Name: Jakarta Mail 2.1
 WLP-AlsoKnownAs: javaMail-2.1
 WLP-Activation-Type: parallel
 IBM-API-Package: \
+  jakarta.activation; type="spec", \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mail-2.1/io.openliberty.mail-2.1.feature]:
```suggestion
```
Put this in io.openliberty.activation.internal-2.1

File: dev/com.ibm.websphere.appserver.features/visibility/public/restfulWSClient-3.1/io.openliberty.restfulWSClient-3.1.feature [hunk 4]
```diff
@@ -2,7 +2,8 @@
 symbolicName=io.openliberty.restfulWSClient-3.1
 visibility=public
 singleton=true
-IBM-API-Package: jakarta.ws.rs; type="spec", \
+IBM-API-Package: jakarta.activation; type="spec", \
+ jakarta.ws.rs; type="spec", \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/public/restfulWSClient-3.1/io.openliberty.restfulWSClient-3.1.feature]:
```suggestion
IBM-API-Package: jakarta.ws.rs; type="spec", \
```

**Review by jhanders34 — CHANGES_REQUESTED** [review]

**Comment by neuwerk** [comment]
#build

File: dev/io.openliberty.org.eclipse.angus.activation/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Import-Package: \
+  *
+
+app-resources= \
+  META-INF/services/jakarta.activation.spi.MailcapRegistryProvider | \
+  META-INF/services/jakarta.activation.spi.MimeTypeRegistryProvider
+
+Service-Component: \
+  io.openliberty.jakarta.activation_2_1_${replace;${bVersion};\\.;_}.ResourceProvider; \
```
Comment (jhanders34) [inline dev/io.openliberty.org.eclipse.angus.activation/bnd.bnd]: Do you want to change this name to match the bundle name?

**Review by jhanders34 — DISMISSED** [review]

**Review by jhanders34 — APPROVED** [review]

**Comment by neuwerk** [comment]
#build

**Review by WhiteCat22 — APPROVED** [review]

**Comment by jhanders34** [comment]
#libby

**Comment by neuwerk** [comment]
Failure is unrelated to these changes, was a infrastructure timeout in one of the FAT child builds.

**Pull request merged**